authoringAssistant/src/main/java/explorableviz/transparenttext/Main.java

```diff
@@ -75,13 +74,22 @@ public static ArrayList<QueryContext> loadTestCases(String testCasesFolder, int
         if (!Files.exists(folder) || !Files.isDirectory(folder)) {
             throw new RuntimeException(STR."Invalid folder path: \{testCasesFolder}");
         }
-        DirectoryStream<Path> directoryStream = Files.newDirectoryStream(folder, Files::isRegularFile);
         ArrayList<QueryContext> queryContexts = new ArrayList<>();
         Random random = new Random(0);
-        for (Path filePath : directoryStream) {
-            TestQueryContext testQueryContext = TestQueryContext.importFromJson(filePath, random);
-            queryContexts.addAll(testQueryContext.instantiate(numInstances));
-        }
+        Files.list(Paths.get(testCasesFolder))
```

Comment:
This looks very similar to the logic you added to `LearningQueryContext.importLearningCaseFromJSON`. When you spot non-trivial patterns like these that get used in multiple places, it’s a good idea to extract a method which can be used in both places.